src/Resources/app/storefront/src/main.js

```diff
@@ -1,6 +1,8 @@
 // Import all necessary Storefront plugins and scss files
 import ConfirmOrderPlugin from './checkout/confirm-order.plugin';
+import AdyenGivingPlugin from './finish/adyen-giving.plugin';
 
 // Register them via the existing PluginManager
 const PluginManager = window.PluginManager;
 PluginManager.register('ConfirmOrderPlugin', ConfirmOrderPlugin, '#adyen-payment-checkout-mask');
+PluginManager.register('AdyenGivingPlugin', AdyenGivingPlugin, '[data-adyen-giving]');
```

Comment:
if you extract the giving block to a separate template, then you can wrap it in a div with a unique ID and then use the ID identifier here similar to how the ConfirmOrderPlugin is registered